### PR TITLE
RDM-12672 - Add attachToCase PATCH endpoint to client library

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

RDM-12672 - Add attachToCase PATCH endpoint to client library and use case-doc-am-client in CCD

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by KleoG — APPROVED** [review]
Looks good, could their potentially be a test for a failing response for attachToCase?

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentsMetadata.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.List;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class CaseDocumentsMetadata {
+
+    @JsonProperty
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentsMetadata.java:17]: There is a @Jacksonized lombok annotation is available now. That eliminates need of @JsonProperty and other lombok annotations like @NoArgsConstructor etc

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/DocumentHashToken.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.hmcts.reform.ccd.document.am.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.validation.annotation.Validated;
+
+@Validated
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/DocumentHashToken.java:10]: may be not need here @Validated

**Comment by smathangi** [comment]
Not related to this PR but could you update method parameter name to 'selfHref' please https://github.com/hmcts/ccd-case-document-am-client/blob/949a898358f36b1b520c71eb82dee6f5758f5f7c/src/main/java/uk/gov/hmcts/reform/ccd/document/am/feign/CaseDocumentClient.java#L92-L93

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/DocumentHashToken.java (hunk 2 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/DocumentHashToken.java:10]: 👍

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentsMetadata.java (hunk 1 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentsMetadata.java:17]: `@Jacksonized was introduced as experimental feature in lombok v1.18.14.`

Are we ok with using experimental features ?

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/DocumentHashToken.java (hunk 2 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/DocumentHashToken.java]: Fixed in 59b143a

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentsMetadata.java (hunk 1 above)
Comment (smathangi) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentsMetadata.java:17]: That's introduced almost a year ago.. if our tests are passing then i think it's OK.

**Review by smathangi — APPROVED** [review]
LGTM

**Pull request merged**